examples/Bickley_jet_shallow_water.jl

```diff
@@ -0,0 +1,244 @@
+# # Shallow Water Example: an unstable Bickley jet 
+#
+# This example shows how to use `Oceananigans.ShallowWaterModel` to simulate
+# the evolution of an unstable, geostrophically balanced, Bickley jet.
+# The model solves the governing equations for the shallow water model in
+# conservation form.  The geometry is that of a periodic channel
+# in the ``x``-direction with a flat bottom and a free-surface.  The initial
+# conditions are that of a Bickley jet with small amplitude perturbations.
+# The interested reader can see ["The nonlinear evolution of barotropically unstable jets," J. Phys. Oceanogr. (2003)](https://doi.org/10.1175/1520-0485(2003)033<2173:TNEOBU>2.0.CO;2)
+# for more details on this specific problem. 
+#
+# Unlike the other models, the fields that are simulated are the mass transports, 
+# ``uh`` and ``vh`` in the ``x`` and ``y`` directions, respectively,
+# and the height ``h``.  Note that ``u`` and ``v`` are the velocities, which 
+# can easily be computed when needed.
+#
+# ## Install dependencies
+#
+# First we make sure that we have all of the packages that are required to
+# run the simulation.
+# ```julia
+# using Pkg
+# pkg"add Oceananigans, NCDatasets, Plots, Printf, Polynomials"
+# ```
+#
+
+using Oceananigans
+using Oceananigans.Models: ShallowWaterModel
+
+# ## Two-dimensional domain 
+#
+# The shallow water model is a two-dimensional model and we must specify
+# the number of vertical grid points to be one.  
+# We pick the length of the domain to fit one unstable mode along the channel
+# exactly.  Note that ``Lz`` is the mean depth of the fluid.  It is determined
+# using linear stability theory that the zonal wavenumber that yields the most unstable
+# mode that fits in this zonal channel has a growth rate of ``\sigma \approx 0.14``.
+
+Lx = 2π
+Ly = 20
+Lz = 1
+Nx = 128
+Ny = Nx
+
+grid = RegularRectilinearGrid(size = (Nx, Ny, 1),
+                            x = (0, Lx), y = (0, Ly), z = (0, Lz),
+                            topology = (Periodic, Bounded, Bounded))
```

Comment:
```suggestion
# The shallow water model is a two-dimensional model and thus the number of vertical grid 
# points `N_z` must be set to one. Note that ``L_z`` is the mean depth of the fluid.

Lx, Ly, Lz = 2π, 20, 1
Nx, Ny, Nz = 128, 128, 1

grid = RegularRectilinearGrid(size = (Nx, Ny, Nz),
                                 x = (0, Lx),
                                 y = (0, Ly),
                                 z = (0, Lz),
                          topology = (Periodic, Bounded, Bounded))
```

Reply:
(I suggest moving the remark about theoretical growth rate a bit further down.)

Reply:
Good idea!

Reply:
Agreed